Index: dpdk/lib/librte_eal/linuxapp/eal/eal_pci_vfio.c
===================================================================
--- dpdk.orig/lib/librte_eal/linuxapp/eal/eal_pci_vfio.c
+++ dpdk/lib/librte_eal/linuxapp/eal/eal_pci_vfio.c
@@ -448,6 +448,7 @@ pci_vfio_get_group_fd(int iommu_group_no
 		/* if the fd is valid, create a new group for it */
 		if (vfio_cfg.vfio_group_idx == VFIO_MAX_GROUPS) {
 			RTE_LOG(ERR, EAL, "Maximum number of VFIO groups reached!\n");
+			close(vfio_group_fd);
 			return -1;
 		}
 		vfio_cfg.vfio_groups[vfio_cfg.vfio_group_idx].group_no = iommu_group_no;
